client/test/explorer/blockchainExplorer.test.ts

```diff
@@ -30,273 +41,523 @@ describe('BlockchainExplorer', () => {
+                            certificatePath: path.join(rootPath, '../../test/data/connectionOne/credentials/certificate'),
+                            privateKeyPath: path.join(rootPath, '../../test/data/connectionOne/credentials/privateKey')
+                        }]
+                };
+
+                connections.push(myConnection);
+
+                await vscode.workspace.getConfiguration().update('fabric.connections', connections, vscode.ConfigurationTarget.Global);
+
+                const blockchainNetworkExplorerProvider = myExtension.getBlockchainNetworkExplorerProvider();
+                const allChildren = await blockchainNetworkExplorerProvider.getChildren();
+
+                allChildren.length.should.equal(2);
+                allChildren.length.should.equal(2);
+
+                const connectionTreeItem = allChildren[0] as ConnectionTreeItem;
+
+                connectionTreeItem.label.should.equal('myConnection');
+                connectionTreeItem.collapsibleState.should.equal(vscode.TreeItemCollapsibleState.Collapsed);
+                connectionTreeItem.connection.should.deep.equal(myConnection);
+                should.not.exist(connectionTreeItem.command);
+                allChildren[1].label.should.equal('Add new network');
+
+                const errorSpy = mySandBox.spy(vscode.window, 'showErrorMessage');
+
+                await blockchainNetworkExplorerProvider.getChildren(connectionTreeItem);
+
+                errorSpy.should.have.been.called;
+            });
+
+            it('should handle error with tree', async () => {
+                await vscode.extensions.getExtension('IBM.blockchain-network-explorer').activate();
+
+                const connections: Array<any> = [];
+
+                const rootPath = path.dirname(__dirname);
+
+                const myConnection = {
+                    name: 'myConnection',
+                    connectionProfilePath: path.join(rootPath, '../../test/data/connectionTwo/connection.json'),
+                    identities: [{
+                        certificatePath: path.join(rootPath, '../../test/data/connectionTwo/credentials/badPath/certificate'),
+                        privateKeyPath: path.join(rootPath, '../../test/data/connectionTwo/credentials/privateKey')
+                    }]
+                };
+
+                connections.push(myConnection);
+
+                await vscode.workspace.getConfiguration().update('fabric.connections', connections, vscode.ConfigurationTarget.Global);
+
+                const blockchainNetworkExplorerProvider = myExtension.getBlockchainNetworkExplorerProvider();
+
+                const errorSpy = mySandBox.spy(vscode.window, 'showErrorMessage');
+
+                mySandBox.stub(blockchainNetworkExplorerProvider, 'createConnectionTree').rejects({message: 'some error'});
+
+                await blockchainNetworkExplorerProvider.getChildren();
+
+                errorSpy.should.have.been.calledWith('some error');
```

Comment:
check we check for a specific error message? Applies to the other errorSpy lines too